### PR TITLE
feat(callout): update quote and media with new tokens

### LINKED ISSUE #9615: [Callout quote] Add color background options
**Follow on to design:** #9572

**[Updated design spec](https://www.figma.com/file/KQdUduecrBf4DCjRZnsl25/Layout-component--callout-quote--visual-specs--2.0.0?node-id=0%3A1)**

**[Figma design exploration](https://www.figma.com/file/jqUls2vqW1RDQbPReeoTaL/Callout-quote-%2B-quote-exploration?node-id=0%3A1)** 

To give `callout quote` more flexibility for the latest brand updates, the component needs added options for color backgrounds. This should be a new attribute that mirrors `quote`'s Color Scheme (color-scheme):
- [ ] regular color scheme (matching the theme ui-background)
- [ ] ui-01 background
- [ ] cyan background (using new callout component color tokens)
- [ ] purple background (using new callout component color tokens)

### LINKED ISSUE #9621: [Callout] Add component color tokens
Similar to Carbon's [tag](https://carbondesignsystem.com/components/tag/style/#color) component, we need to add component-specific color tokens to handle the cyan and purple options for the `callout` components: `callout with media` and `callout quote`.

The new color tokens would map to the IDL colors below based on theme:

`$callout-background-cyan`: 
- White and gray10 use → cyan-10
- gray90 and gray100 use → cyan-90

`$callout-color-cyan`: 
- White and gray10 use → cyan-80
- gray90 and gray100 use → cyan-10

`$callout-caption-cyan`: 
- White and gray10 use → cyan-70
- gray90 and gray100 use → cyan-30

`$callout-background-purple`: 
- White and gray10 use → purple-10
- gray90 and gray100 use → purple-90

`$callout-color-purple`: 
- White and gray10 → purple-80
- gray90 and gray100 → purple-10

`$callout-caption-purple`: 
- White and gray10 use → purple-70
- gray90 and gray100 use → purple-30

### LINKED ISSUE #10725: [Component] Callout quote

### LINKED ISSUE #10847: [Component] Quote

### LINKED ISSUE #10848: [Component] Horizontal rule
From #10139 : 
- [ ] For contrast, the `medium-contrast` option is deprecated in v2
- [ ] Also for contrast, the naming of the remaining two contrast values has been adjusted: `low-contrast` is now `subtle`, and `high-contrast` is now `strong`. This is to align with Carbon core naming.

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

Closes #10725 
Closes #9615
Closes #9621 
Closes #10848
Closes #10847

### Description

This adds the new callout tokens and updates callout quote which uses horizontal rule and quote, which have also been updated. I did not completely update `callout-with-media` yet, but it does include the new tokens.

### Changelog

**New**

- callout tokens

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

File: docs/dotcom-v2-migration.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ For Carbon v11 migration guidance, see their
 | back-to-top               | View changes [here](#backtotop)        |
 | background-media          | No API changes.                        |
 | button                    | This component is deprecated in v2 in favor for Carbon's core Button |
+| callout-quote             | View changes [here](#calloutQuote)          |
```
Comment (emyarod) [inline docs/dotcom-v2-migration.md:21]:
```suggestion
| callout-quote             | View changes [here](#callout-quote)          |
```

File: docs/dotcom-v2-migration.md [hunk 2]
```diff
@@ -45,6 +46,11 @@ For Carbon v11 migration guidance, see their
 
 - 
 
+### CalloutQuote
```
Comment (emyarod) [inline docs/dotcom-v2-migration.md:49]:
```suggestion
### Callout quote
```

File: packages/styles/scss/components/callout-with-media/_callout-with-media.scss [hunk 3]
```diff
@@ -11,54 +11,105 @@
...
     max-width: rem(640px);
     width: auto;
   }
 
-  .#{$prefix}--callout-with-media .#{$prefix}--callout__content {
-    padding-top: 0;
-  }
+  // .#{$prefix}--callout-with-media .#{$prefix}--callout__content {
+  //   padding-top: 0;
+  // }
```
Comment (emyarod) [inline packages/styles/scss/components/callout-with-media/_callout-with-media.scss:112]:
```suggestion
```

**Review by emyarod — COMMENTED** [review]
are we set on using camel case for the migration doc? if we are then you can ignore my suggestions there!

File: docs/dotcom-v2-migration.md (hunk 1 above)
Comment (ariellalgilmore) [reply docs/dotcom-v2-migration.md:21]: oh good point!

**Review by RichKummer — COMMENTED** [review]
Hey @ariellalgilmore, this is looking great! Couple of notes:

1. The `link with icon` looks like it's coming in as `$link-inverse` across all color schemes. It should be `$link-primary` for all of them.
Screenshot 2023-08-31 at 2 01 22 PM


2. This might be a Storybook issue, but for color-scheme `none` – can this be renamed to `regular` to match similar components?

3. For the currently named `gray10` color scheme, can we change that to `layer`? I just want to make sure that adopters know it's not gray10 for every theme.

4. For cyan, it looks like cyan 80 is being brought in. Should be cyan-90 where the [story](https://github.com/orgs/carbon-design-system/projects/56/views/1?filterQuery=-label%3Aicebox%2Cinactive+callout&pane=issue&itemId=29290424) lists. Applies to both copy and background.

5. Same as above with purple. Should be `purple-90` not `purple-80`


6. It looks like the medium-contrast issue is linked for `horizontal rule`, but it looks like it's still in Storybook?
Screenshot 2023-08-31 at 2 35 52 PM

**Comment by RichKummer** [comment]
One last thing: we should still keep the inverse option for adopters, but the `regular` color scheme should become the new default. I don't think I saw the inverse style option in the list but maybe that's just Storybook missing it.

**Comment by ariellalgilmore** [comment]
@RichKummer the `color-scheme` property was updated to include `inverse` and change `gray10` to `layer`. The cyan/purple tokens were updated as well. `hr` was also updated with the new `constrast` types: `subtle` and `strong`

**Review by RichKummer — COMMENTED** [review]
@ariellalgilmore all the color token tweaks and the `horizontal rule` updates look solid. I did find one odd bug in `callout with media`:

1. I'm seeing some weird added spacing between the body copy and the media. See below

**This PR:**
Screenshot 2023-09-01 at 3 06 42 PM

**Current WC Storybook:**
Screenshot 2023-09-01 at 3 06 49 PM

The spacing between the body copy and media render as `32px` across breakpoints.

**Comment by ariellalgilmore** [comment]
Thanks @RichKummer! i was going to open another pr later with callout-with-media seems like there's something weird with video as well

**Review by RichKummer — APPROVED** [review]
The core updates are here, approving and Ari is opening a PR to update the odd spacing in `callout media`

File: packages/styles/scss/components/callout-quote/_callout-quote.scss [hunk 4]
```diff
@@ -53,4 +35,70 @@
...
+  }
+
+  :host(#{$c4d-prefix}-callout-link-with-icon[color-scheme='inverse']) {
+    a.#{$prefix}--link-with-icon.#{$prefix}--link.#{$prefix}--link-with-icon__icon-right {
+      color: $link-inverse;
+
+      &:hover {
+        // @carbon/styles has no hover color for $inverse-link.
+        // Solution is to use the same token and add underline
+        color: $link-inverse;
```
Comment (IgnacioBecerra) [inline packages/styles/scss/components/callout-quote/_callout-quote.scss:86]:
```suggestion
        color: $link-inverse;
        text-decoration: underline;
```
Should we also add the underline here?

File: packages/styles/scss/components/callout-quote/_callout-quote.scss [hunk 5]
```diff
@@ -53,4 +35,70 @@
...
+
+    display: inline-block;
+    a.#{$prefix}--link-with-icon.#{$prefix}--link.#{$prefix}--link-with-icon__icon-right {
+      color: $link-primary;
+      outline-color: $focus-inverse;
+
+      &:hover {
+        // @carbon/styles has no hover color for $inverse-link.
+        // Solution is to use the same token and add underline
+        color: $link-primary;
```
Comment (IgnacioBecerra) [inline packages/styles/scss/components/callout-quote/_callout-quote.scss:69]:
```suggestion
        color: $link-inverse;
```
Should this be changed?

File: packages/web-components/src/components/callout-quote/callout-quote.ts [hunk 6]
```diff
@@ -7,21 +7,63 @@
...
+import { property } from 'lit/decorators.js';
+import { COLOR_SCHEME } from '../../component-mixins/callout/defs';
+import C4DCalloutLinkWithIcon from './callout-link-with-icon';
 
-const { stablePrefix: c4dPrefix } = settings;
+const { prefix, stablePrefix: c4dPrefix } = settings;
 
 /**
- * Callout Data.
+ * Callout Qupte.
```
Comment (IgnacioBecerra) [inline packages/web-components/src/components/callout-quote/callout-quote.ts:23]:
```suggestion
 * Callout Quote.
```

**Review by IgnacioBecerra — COMMENTED** [review]
This looks great, good work @ariellalgilmore! Just a couple things:

File: packages/styles/scss/components/callout-quote/_callout-quote.scss (hunk 5 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/callout-quote/_callout-quote.scss]: ooh good catch i updated to use `link-primary-hover` actually which was added as a token!

File: packages/styles/scss/components/callout-quote/_callout-quote.scss (hunk 4 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/callout-quote/_callout-quote.scss:82]: i actually removed underline from both... there's already styles adding the underline from other imported styles, so the underline still is visible

**Review by IgnacioBecerra — APPROVED** [review]
LGTM!! Thanks @ariellalgilmore!

**Pull request merged**